.travis.yml

```diff
@@ -7,6 +7,9 @@ env:
   - ARMADILLO=minimum
 
 before_install:
+  - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
```

Comment:
Personally I would put this to the end, after the build and test step; let's say I have a minor spelling fix, that would probably not end up in the history, but I still like to check that I haven't introduced anything that breaks the build.

Reply:
@zoq the reason I put it first instead of last is two-fold:

1. Less likely to see it at the "end" of a raw log from travis 
   - ~3k lines each section is collapsed though.
2. The total time spent on CI is significant.
   - Likely to correct immediate if flagged earlier vs. later in the CI process. 
   - In particular, CI fails in ~2 minutes vs. ~14 min 49 sec.

Happy to change check position depending on what the consensus is.

Reply:
I see what you mean, do you think we should create another job on jenkins `sanity check` or something, that way get a result immediately and we don't have to scroll through a bunch of console output. If someone commits a history commit right after he notice that the jenkins job fails the travis job should be automatically canceled.

Reply:
@zoq I'm not familiar with Jenkins configuration. So, I'm deferring to your judgement and, potentially, a helping hand in configuring Jenkins. 

Though, to be honest, I'm frequently checking Travis over Jenkins...

Reply:
You could accomplish this by adding another line to the env section, and checking if _that_ is set. That way, we'd get both versions of armadillo and the sanity check.
I can't seem to suggest on multiple lines, but something like:
```env:
  - ARMADILLO=latest
  - ARMADILLO=minimum
 - SANITY=history

before_install:
  - if [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ "$SANITY" != "history" ]; then
```

Reply:
That sounds like a great idea to me.

Reply:
@zoq done. See: 09291ce

@birm thanks!

Reply:
Ah, it looks like you'd want to exit 0 on an ok check too, since it's trying to test everything right now.